### PR TITLE
feat: add `ReadFileWithSizeLimit` and use it on `x/wasm`

### PR DESCRIPTION
**Pull request by tnasu**
## Description

Reading a huge file has the risk of the crash application by using `ioutil.ReadFile/os.ReadFile`.
Here is the updates:
* Add `ReadFileWithSizeLimit`
  * Expanded: https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/os/file.go;l=672
* Use it in:
  * x/wasm/GenesisStoreCodeCmd
  * x/wasm/ProposalStoreCodeCmd
  * x/wasm/StoreCodeCmd
  * x/wasm/StoreCodeAndInstantiateContractCmd
  * x/distribution/GetCmdSubmitProposal

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

**Review by dudong2 — APPROVED** [review]
LGTM

**Review by zemyblue — APPROVED** [review]

**Comment by tnasu** [comment]
Add to address it on x/distribution
* https://github.com/line/lbm-sdk/pull/696/commits/7b4bb6eb543aaa784466ab11c0004a05e6ecde23

File: CHANGELOG.md [hunk 1]
```diff
@@ -73,6 +73,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (refactor) [\#685](https://github.com/line/lbm-sdk/pull/685) remove x/foundation UpdateValidatorAuthsProposal
 * (x/foundation) [\#686](https://github.com/line/lbm-sdk/pull/686) remove `Minthreshold` and `MinPercentage` from x/foundation config
 * (x/foundation) [\#693](https://github.com/line/lbm-sdk/pull/693) add pool to the state of x/foundation
+* (x/wasm) [\#696](https://github.com/line/lbm-sdk/pull/696) x/wasm - add checking a wasm file size before reading it
+* (x/distribution) [\#696](https://github.com/line/lbm-sdk/pull/696) x/distribution - add checking a proposal file size before reading it
```
Comment (zemyblue) [inline CHANGELOG.md:77]: How about changing one line?

Comment (tnasu) [reply CHANGELOG.md]: I've updated it. Thanks

**Review by zemyblue — APPROVED** [review]

**Review by dudong2 — APPROVED** [review]

**Pull request merged**